File: front/apps/web-antd/src/api/feedalyze/customer.ts
/**
 * 客户管理 API
 */

import { requestClient } from '#/api/request';

/** 客户对象 */
export interface Customer {
  id: string;
  tenant_id: string;
  name: string;
  customer_type: string;
  business_value: number;
  created_time: string;
  updated_time: string;
  deleted_at?: string;
}

/** 创建客户参数 */
export interface CreateCustomerParams {
  name: string;
  customer_type?: string;
  business_value?: number;
}

/** 更新客户参数 */
export interface UpdateCustomerParams {
  name?: string;
  customer_type?: string;
  business_value?: number;
}

/** 客户列表查询参数 */
export interface CustomerListParams {
  skip?: number;
  limit?: number;
}

/**
 * 获取客户列表
 */
export async function getCustomerList(params: CustomerListParams) {
  return requestClient.get<Customer[]>('/feedalyze/customers', { params });
}

/**
 * 创建客户
 */
export async function createCustomer(data: CreateCustomerParams) {
  return requestClient.post<Customer>('/feedalyze/customers', data);
}

/**
 * 更新客户
 */
export async function updateCustomer(id: string, data: UpdateCustomerParams) {
  return requestClient.put<Customer>(`/feedalyze/customers/${id}`, data);
}

/**
 * 删除客户
 */
export async function deleteCustomer(id: string) {
  return requestClient.delete(`/feedalyze/customers/${id}`);
}

/** 客户类型选项 */
export const CUSTOMER_TYPES = [
  { value: 'normal', label: '普通客户', business_value: 1 },
  { value: 'paid', label: '付费客户', business_value: 3 },
  { value: 'major', label: '大客户', business_value: 5 },
  { value: 'strategic', label: '战略客户', business_value: 10 },
];

File: server/backend/app/userecho/api/v1/customer.py
"""客户 API 端点"""

from fastapi import APIRouter

from backend.app.userecho.schema.customer import CustomerCreate, CustomerUpdate
from backend.app.userecho.service import customer_service
from backend.common.response.response_code import CustomResponse
from backend.common.response.response_schema import response_base
from backend.common.security.jwt import CurrentTenantId
from backend.database.db import CurrentSession

router = APIRouter(prefix='/customers', tags=['UserEcho - 客户管理'])


@router.get('', summary='获取客户列表')
async def get_customers(
    db: CurrentSession,
    tenant_id: str = CurrentTenantId,
    skip: int = 0,
    limit: int = 100,
):
    """获取客户列表"""
    customers = await customer_service.get_list(db=db, tenant_id=tenant_id, skip=skip, limit=limit)
    return response_base.success(data=customers)


@router.post('', summary='创建客户')
async def create_customer(
    data: CustomerCreate,
    db: CurrentSession,
    tenant_id: str = CurrentTenantId,
):
    """
    创建客户

    - **name**: 客户名称
    - **customer_type**: 客户类型 (normal/paid/major/strategic)
    - **business_value**: 商业价值权重 (1-10)
    """
    customer = await customer_service.create_customer(db=db, tenant_id=tenant_id, data=data)
    return response_base.success(data=customer)


@router.put('/{customer_id}', summary='更新客户')
async def update_customer(
    customer_id: str,
    data: CustomerUpdate,
    db: CurrentSession,
    tenant_id: str = CurrentTenantId,
):
    """更新客户"""
    customer = await customer_service.update_customer(db=db, tenant_id=tenant_id, customer_id=customer_id, data=data)
    if not customer:
        return response_base.fail(res=CustomResponse(code=400, msg='客户不存在'))
    return response_base.success(data=customer)


@router.delete('/{customer_id}', summary='删除客户')
async def delete_customer(
    customer_id: str,
    db: CurrentSession,
    tenant_id: str = CurrentTenantId,
):
    """删除客户（软删除）"""
    success = await customer_service.delete_customer(db=db, tenant_id=tenant_id, customer_id=customer_id)
    if not success:
        return response_base.fail(res=CustomResponse(code=400, msg='客户不存在或已删除'))
    return response_base.success(res=CustomResponse(code=200, msg='删除成功'))
